crypto/jwks/component.go

```diff
@@ -94,11 +100,20 @@ func (k *jwksCrypto) Features() []contribCrypto.Feature {
 }
 
 // Init the JWKS object from the metadata property
-func (k *jwksCrypto) initJWKS(ctx context.Context) error {
+func (k *jwksCrypto) initJWKS() error {
 	if len(k.md.JWKS) == 0 {
 		return errors.New("metadata property 'jwks' is required")
 	}
 
+	// Use context based on close channel.
+	ctx, cancel := context.WithCancel(context.Background())
+	k.wg.Add(1)
+	go func() {
```

Comment:
The problem with not using a component-scoped context is that now we have an extra goroutine (which apparently is a big deal lately :) )